src/config.rs

```diff
@@ -295,7 +295,9 @@ impl Config {
     ///
     /// For this, some values need to be converted to other types and some
     /// defaults need to be set (sometimes based on env variables).
-    fn from_raw(raw_config: RawConfig) -> Result<Self> {
+    fn from_raw(config_file_path: PathBuf, raw_config: RawConfig) -> Result<Self> {
+        // Safe to unwrap since we managed to read the file
+        let config_dir_path = config_file_path.parent().unwrap();
```

Comment:
I think this should be
```suggestion
    fn from_raw(raw_config: RawConfig, relative_path_root: &Path) -> Result<Self> {
```
where `relative_path_root` replaces `config_dir_path` - the caller of `from_raw` should perform the `config_file_path.parent()` call, because they are the ones who know that they got the `RawConfig` from this path and thus it exists. We here shouldn't really know how `config_file_path` and `raw_config` are related

This also eliminates the need for the added comments in this function, in my opinion

Reply:
Fair point, but if we pass the base dir and not config path, someone reading the code might start wondering "Hmm, and where are the paths resolved from? Is it always relative to the config file? Do I need to check for that?". If we pass the config file path, it's more obvious what the function does, imo.

Reply:
Hm, I see, this makes `from_raw` more abstract than it is right now. I think that this abstraction is correct though, because we already moved "reading the raw config from the file" to another place, so we would reintroduce this knowledge here. This was what tripped me up, because the comment also says "we know we read the file, so we can unwrap", but in the context of just this method, we don't actually know that, we are just lifting the raw config to a proper Config object. Additionally, there isn't really a reason to limit what the path argument should be - we only use it as a root for the relative paths

What do you think?

Reply:
Hmm, thinking about it a bit more, the comment should be more like "if unwrap fails, it's the caller's fault, since they should pass a file path".

Your point about abstraction makes sense – if someone wanted to e.g. support reading config from stdin and resolve relative to the working directory, we'd want to pass in a directory, and technically, relative paths are always resolved relative to a directory, not to a file (although it's a little bit confusing for the config file scenario). Do not have a strong preference, so let's go with your suggestion.

Reply:
Updated.